osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/infra/networking/DlmsPushNotificationDecoder.java

```diff
@@ -130,28 +315,80 @@ private DlmsPushNotification buildPushNotification() {
         }
     }
 
-    private void decodeObisCodeData(final ChannelBuffer buffer) {
+    private void decodeObisCodeData(final ChannelBuffer buffer, Protocol protocol, boolean alarmRegisterExpected) throws UnrecognizedMessageDataException {
 
         final byte[] logicalNameBytes = new byte[NUMBER_OF_BYTES_FOR_LOGICAL_NAME];
         buffer.readBytes(logicalNameBytes, 0, NUMBER_OF_BYTES_FOR_LOGICAL_NAME);
 
-        if (Arrays.equals(SMS_OBISCODE_BYTES, logicalNameBytes)) {
-            this.builder.withTriggerType(PUSH_SMS_TRIGGER);
-        } else if (Arrays.equals(CSD_OBISCODE_BYTES, logicalNameBytes)) {
-            LOGGER.warn("CSD Push notification not supported");
-            this.builder.withTriggerType(PUSH_CDS_TRIGGER);
-        } else if (Arrays.equals(SCHEDULER_OBISCODE_BYTES, logicalNameBytes)) {
-            LOGGER.warn("Scheduler Push notification not supported");
-            this.builder.withTriggerType(PUSH_SCHEDULER_TRIGGER);
-        } else {
-            LOGGER.warn("Unknown Push notification not supported. Unable to decode");
-            this.builder.withTriggerType("");
+        try {
+            if (!alarmRegisterExpected && isLogicalNameSmsTrigger(logicalNameBytes, protocol)) {
+                this.builder.withTriggerType(PUSH_SMS_TRIGGER);
```

Comment:
This is not the expected way to use a builder. I don't see the advantage here since it is not helping to make the construction of an instance more intuitive or avoiding a constructor with a large number of arguments. 

I suggest to get rid of the builder and just manipulate the fields.

Code like `builder.appendBytes(...)` already indicates that the builder pattern is not really a match for this use case IMO.

Reply:
It avoids having to pass arguments multiple times. Maybe it's not strictly a builder pattern, but I'd like to keep it.

Reply:
If you're happy with it... I still think it looks weird.